packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -489,11 +489,44 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       assert(minScrollExtent != null);
       assert(maxScrollExtent != null);
       assert(minScrollExtent <= maxScrollExtent);
+      final ScrollMetrics oldPosition = haveDimensions ? copyWith() : null;
```

Comment:
Is this truly the "old position"?  For example what's the value of `pixels`?  It's not the value it was previously, it is what the value is currently.  Do you think it would be more clear to pass the old extents into the function instead of a ScrollMetric?  There is extra baggage with a ScrollMetric that might be weird referring to this as "oldPosition" since it is the current position modulo minScrollExtent and  maxScrollExtent?

Reply:
It also includes useful information like the axisDirection and the viewportDimension, but you're right that only the min/max scroll extents can be different between the two arguments.

I'm loathe to make `adjustPositionForNewDimensions` take 9 arguments or whatever it would be to pass all the useful information along, though. I guess I could do 5 arguments (oldMinScrollExtent, oldMaxScrollExtent, metrics, isScrolling, velocity) but then the code looks weird because it's doing things like "oldMinScrollExtent < metrics.pixels" and comparing that to "metrics.minScrollExtent < metrics.pixels" or whatever, which is asymmetric and makes the code harder to read...

I called it "old" and "new" "position" because it's literally copied from the ScrollPosition. Would it help to document the weirdness here?

Reply:
You can keep the same number of arguments if you send in a Range datatype.
`adjustPositionForNewExtents(this, Range(oldMinExtents, oldMaxExtents), isScrolling,  velocity)`

You could even compress it smaller if you made an immutable view for ScrollActivity and passed the current activity instead of the isScrolling and velocity (I'm not sure if it is  immutable already): `adjustPositionForNewExtents(this, Range(oldMinExtents, oldMaxExtents), activity)`

Reply:
that would require introducing a whole lot more API surface, though. So while we simplify this one API, we do it at the cost of more code size and a more complicated API overall.

Reply:
I'm surprised we don't have a Range class already, pretty typical and useful.  I think my suggestion is more clear and I'd error on the side of clear API over code size when we are just adding something like a Range class.  It's not worth a long discussion, if you don't think it's worth doing that's fine.

Reply:
I poked at this some more but couldn't come up with something that would improve it enough to outweigh the costs. :-( I'll land this like this for now and if we can come up with something better we can always change it. This is unlikely to be widely used anyway.